zone/gm_commands/parcels.cpp

```diff
@@ -0,0 +1,264 @@
+				c->Message(
+					Chat::Yellow, "Your quantity of {} copper pieces was too large.  Set to max quantity of {}.",
+					quantity,
+					INT32_MAX
+				);
+				quantity = INT32_MAX;
+			}
+			auto item = database.GetItem(PARCEL_MONEY_ITEM_ID);
+			if (!item) {
+				c->Message(Chat::Yellow, "Could not find item with id {}", item_id);
+				return;
+			}
+
+			auto inst = database.CreateItem(item, 1);
+			if (!inst) {
+				c->Message(Chat::Yellow, "Could not find item with id {}", item_id);
+				return;
+			}
+
+			auto money = inst->DetermineMoneyStringForParcels(quantity == 0 ? 1 : quantity);
+
+			ParcelsRepository::Parcels parcel_out;
+			parcel_out.from_name = c->GetName();
+			parcel_out.note      = note.empty() ? "" : note;
+			parcel_out.sent_date = time(nullptr);
+			parcel_out.quantity  = quantity == 0 ? 1 : quantity;
+			parcel_out.item_id   = PARCEL_MONEY_ITEM_ID;
+			parcel_out.to_name   = to_name;
+			parcel_out.slot_id   = next_slot;
+			parcel_out.id        = 0;
+
+			auto result = ParcelsRepository::InsertOne(database, parcel_out);
+			if (!result.id) {
+				LogError("Failed to add parcel to database.  From {} to {} item {} quantity {}", parcel_out.from_name,
+						 parcel_out.to_name, parcel_out.item_id, parcel_out.quantity);
+				c->Message(Chat::Yellow, "Unable to save parcel to the database. Please contact an administrator.");
+				return;
+			}
+
+			c->MessageString(
+				Chat::Yellow, PARCEL_DELIVERY, c->GetCleanName(), money.c_str(),
+				send_to_client.at(0).character_name.c_str());
+
+			if (player_event_logs.IsEventEnabled(PlayerEvent::PARCEL_SEND)) {
+				PlayerEvent::ParcelSend e{};
+				e.from_player_name = parcel_out.from_name;
+				e.to_player_name   = parcel_out.to_name;
+				e.item_id          = parcel_out.item_id;
+				e.quantity         = parcel_out.quantity;
+				e.sent_date        = parcel_out.sent_date;
+
+				RecordPlayerEventLogWithClient(c, PlayerEvent::PARCEL_SEND, e);
+			}
+
+			Parcel_Struct ps{};
+			ps.item_slot = parcel_out.slot_id;
+			strn0cpy(ps.send_to, parcel_out.to_name.c_str(), sizeof(ps.send_to));
+
+			c->SendParcelDeliveryToWorld(ps);
+		}
+		else {
+			auto item = database.GetItem(item_id);
+			if (!item) {
+				c->Message(Chat::Yellow, "Could not find an item with id {}", item_id);
+				return;
+			}
+
+			auto inst = database.CreateItem(item, quantity > INT16_MAX ? INT16_MAX : (int16) quantity);
+			if (!inst) {
+				c->Message(Chat::Yellow, "Could not find an item with id {}", item_id);
+				return;
+			}
+
+			if (inst->IsStackable()) {
+				quantity = quantity > inst->GetItem()->StackSize ? inst->GetItem()->StackSize : (int16) quantity;
+			}
+			else if (inst->GetItem()->MaxCharges > 0) {
+				quantity = quantity >= inst->GetItem()->MaxCharges ? inst->GetItem()->MaxCharges : (int16) quantity;
+			} else {
+				quantity = 1;
+			}
+
+			ParcelsRepository::Parcels parcel_out;
+			parcel_out.from_name = c->GetName();
+			parcel_out.note      = note.empty() ? "" : note;
```

Comment:
Don't think we need to do this, since we're setting to empty if it's empty it shouldn't matter, right?

Reply:
Resolved